memcached/memcached_1.6.19.patch

```diff
@@ -0,0 +1,1308 @@
+                     if (ssl == NULL) {
+                         if (settings.verbose) {
+@@ -3060,6 +3091,7 @@ static void drive_machine(conn *c) {
+                         close(sfd);
+                         break;
+                     }
++#ifdef TLS
+                     SSL_set_fd(ssl, sfd);
+                     int ret = SSL_accept(ssl);
+                     if (ret <= 0) {
+@@ -3076,6 +3108,45 @@ static void drive_machine(conn *c) {
+                             break;
+                         }
+                     }
++#elif WOLFSSL_MEMCACHED
++                    wolfSSL_set_fd(ssl, sfd);
++                    int ret = wolfSSL_accept(ssl);
++                    if (ret != WOLFSSL_SUCCESS) {
++                        int err = wolfSSL_get_error(ssl, ret);
++                        if (err != WOLFSSL_ERROR_WANT_READ && err != WOLFSSL_ERROR_WANT_WRITE) {
++                            if (settings.verbose) {
++                                fprintf(stderr, "SSL connection failed with error code : %d : %s\n",
++                                        err, wolfSSL_ERR_reason_error_string(err));
++                            }
++                            wolfSSL_free(ssl);
++                            close(sfd);
++                            STATS_LOCK();
++                            stats.ssl_handshake_errors++;
++                            STATS_UNLOCK();
++                            break;
++                        }
++                        else {
++                            /* Unlike OpenSSL, wolfSSL does not automatically retry
++                               on non-blocking connections, the application must
++                               repeat the call until it succeeds when WANT_READ/WANT_WRITE
++                               is returned. */
++                            int maxRetries = 10;
++                            do {
++                                struct pollfd to_poll[1];
++
++                                ret = wolfSSL_accept(ssl);
++                                err = wolfSSL_get_error(ssl, ret);
++                                if (err == WOLFSSL_ERROR_WANT_READ) {
++                                    to_poll[0].fd = sfd;
++                                    to_poll[0].events = POLLIN;
++                                    poll(to_poll, 1, 500);
++                                }
++                                maxRetries--;
++                            } while ((err == WOLFSSL_ERROR_WANT_READ || err == WOLFSSL_ERROR_WANT_WRITE)
++                                     && maxRetries > 0);
++                        }
++                    }
++#endif
+                 }
+                 ssl_v = (void*) ssl;
+ #endif
+@@ -3640,7 +3711,7 @@ static int server_socket(const char *interface,
+                 fprintf(stderr, "failed to create listening connection\n");
+                 exit(EXIT_FAILURE);
+             }
+-#ifdef TLS
++#if defined(TLS) || defined(WOLFSSL_MEMCACHED)
```

Comment:
Would it not be easier to just define `TLS` when `WOLFSSL_MEMCACHED` is defined?

Reply:
Ok that makes sense.